sinaloa/src/ec2_instance.rs

```diff
@@ -0,0 +1,317 @@
+        let privkey_path: &Path = Path::new(&aws_private_key_filename);
+        session.userauth_pubkey_file("ec2-user",
+            None,
+            privkey_path,
+            None
+            )
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+        let mut channel = session.channel_session()
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+        channel.exec(command)
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+        let mut s = String::new();
+        channel.read_to_string(&mut s)
+            .map_err(|err| EC2Error::IOError(err))?;
+        channel.wait_close()
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+        let exit_status = channel.exit_status()
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+        if exit_status != 0 {
+            println!("EC2Instance::excute_command SSH2 Session returned with non-zero exit-status:{:?}", exit_status);
+            return Err(EC2Error::CommandNonZeroStatus(exit_status));
+        }
+        Ok(())
+    }
+
+    pub fn upload_file(&self, filename: &str, dest: &str) -> Result<(), EC2Error> {
+
+        let file_data: Vec<u8> = self.read_file(filename)
+            .map_err(|err| {
+                println!("EC2Instance::upload_file failed to read file:{:?}, received error:{:?}", filename, err);
+                err
+            })?;
+        let full_ip = format!("{:}:{:}", self.private_ip, 22);
+        let tcp = TcpStream::connect(full_ip.clone())
+            .map_err(|err| {
+                println!("EC2Instance::upload_file Failed to connect to EC2instance:{:?}", err);
+                EC2Error::IOError(err)
+            })?;
+
+        let mut session: Session = Session::new().unwrap();
+        session.set_tcp_stream(tcp);
+        session.handshake()
+            .map_err(|err| {
+                println!("EC2Instance::upload_file failed handshake:{:?}", err);
+                EC2Error::SSH2Error(err)
+            })?;
+
+        let (key, key_type) = match session.host_key() {
+            Some((k, kt)) => (k, kt),
+            None => return Err(EC2Error::NoHostKeyError),
+        };
+
+        let mut known_hosts = session.known_hosts()
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+        known_hosts.add(&full_ip, key, &full_ip, key_type.into())
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+
+        let aws_private_key_filename = env::var("AWS_PRIVATE_KEY_FILENAME").expect("Failed to read AWS_PRIVATE_KEY_FILENAME environment variable.");
+        let privkey_path: &Path = Path::new(&aws_private_key_filename);
+        session.userauth_pubkey_file("ec2-user",
+            None,
+            privkey_path,
+            None
+            )
+            .map_err(|err| EC2Error::SSH2Error(err))?;
+
+        let mut remote_file = session.scp_send(Path::new(dest), 0o777, file_data.len() as u64, None)
```

Comment:
Maybe factor out `0o777` out into a constant?

Reply:
...though thinking again 777 is pretty well-known...

Reply:
Other opinions? @geky @ShaleXIONG @basmaelgaabouri @nspin

Reply:
Should it be configurable? If not 0o777 is clearer to me than needing to jump to a constant.